notes/zrq/20210727-01-medium-04-deploy.txt

```diff
@@ -0,0 +1,1415 @@
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+#zrq-notes-time
+#zrq-notes-indent
+#zrq-notes-crypto
+#zrq-notes-ansible
+#zrq-notes-osformat
+#zrq-notes-zeppelin
+#
+
+
+    Target:
+
+        Test deploy of the latest config from git.
+
+    Result:
+
+    Result:
+
+        RandomForest        -
+        MeanProperMotions   -
+        HealpixSourceCounts -
+
+        KinematicClustering -
+
+
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+# Fetch any new tags from the upstream (WFAU) repository
+# and checkout our deployment tag.
+#[user@desktop]
+
+    source "${HOME:?}/aglais.env"
+    pushd "${AGLAIS_CODE}"
+
+        git fetch upstream --tags
+
+        git checkout '20210727-zrq-deploy'
+
+    popd
+
+
+    >   From github.com:wfau/aglais
+    >    * [new tag]         20210727-zrq-deploy -> 20210727-zrq-deploy
+
+
+    >   Note: switching to '20210727-zrq-deploy'.
+    >   
+    >   You are in 'detached HEAD' state. You can look around, make experimental
+    >   changes and commit them, and you can discard any commits you make in this
+    >   state without impacting any branches by switching back to a branch.
+    >   
+    >   ....
+    >   ....
+    >   
+    >   HEAD is now at 38a48db Merge pull request #530 from stvoutsin/issue-multi-user
+
+# -----------------------------------------------------
+# Create a container to work with.
+#[user@desktop]
+
+    source "${HOME:?}/aglais.env"
+
+    podman run \
+        --rm \
+        --tty \
+        --interactive \
+        --name ansibler \
+        --hostname ansibler \
+        --publish 3000:3000 \
+        --publish 8088:8088 \
+        --env "SSH_AUTH_SOCK=/mnt/ssh_auth_sock" \
+        --volume "${SSH_AUTH_SOCK}:/mnt/ssh_auth_sock:rw,z" \
+        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
+        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
+        atolmis/ansible-client:2020.12.02 \
+        bash
+
+
+# -----------------------------------------------------
+# Set the target cloud.
+#[root@ansibler]
+
+    cloudname=gaia-dev
+
+
+# -----------------------------------------------------
+# Delete everything.
+#[root@ansibler]
+
+    time \
+        /deployments/openstack/bin/delete-all.sh \
+            "${cloudname:?}"
+
+    >   real    3m13.010s
+    >   user    1m9.463s
+    >   sys     0m9.187s
+
+
+# -----------------------------------------------------
+# Create everything, using a standard config.
+#[root@ansibler]
+
+    time \
+        /deployments/hadoop-yarn/bin/create-all.sh \
+            "${cloudname:?}" \
+            'cclake-medium-04'
```

Comment:
Are we switching the live service to the "cclake-medium-04" flavour from "medium-04"? Am I right to assume that the amount of Memory & CPU is slightly different or no? In which case should there not be a separate test/issue for finding the right Spark & Yarn configuration before we switch the live service to this?

Reply:
The current live service already is using the `cclake` flavors.
Yes we do need to update the Spark and Yarn configurations to match.